src/lib/components/StyleSelector/StyleSelector.js

```diff
@@ -15,15 +15,31 @@ const mapStateToProps = (state, ownProps) => {
 }
 
 export class StyleSelector extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      styles: props.styles,
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      styles: nextProps.styles,
+    });
+  }
 
   changeStyle = (e) => {
     const style = e.target.value;
     this.props.dispatch(Actions.changeStyle(this.props.mapId, style));
   }
 
   render() {
-    const { styles, mapId } = this.props;
+    const { mapId } = this.props;
+    const { styles } = this.state;
     const sidebarOffset = '0';
+    if (!styles) {
+      return false;
```

Comment:
I thought React will throw errors if false is returned in the render function, should be `null` or like an empty `<div/>`

Reply:
got it, i have addressed